Source/Core/Core/AchievementManager.cpp

```diff
@@ -548,9 +811,12 @@ AchievementManager::ResponseType AchievementManager::FetchGameData()
   std::string api_token = Config::Get(Config::RA_API_TOKEN);
   rc_api_fetch_game_data_request_t fetch_data_request = {
       .username = username.c_str(), .api_token = api_token.c_str(), .game_id = m_game_id};
-  return Request<rc_api_fetch_game_data_request_t, rc_api_fetch_game_data_response_t>(
-      fetch_data_request, &m_game_data, rc_api_init_fetch_game_data_request,
-      rc_api_process_fetch_game_data_response);
+  {
+    std::lock_guard lg{m_lock};
+    return Request<rc_api_fetch_game_data_request_t, rc_api_fetch_game_data_response_t>(
+        fetch_data_request, &m_game_data, rc_api_init_fetch_game_data_request,
+        rc_api_process_fetch_game_data_response);
```

Comment:
This is a network request under the lock and will block everyone else until the request is done. Consider something like:

```
  rc_api_fetch_game_data_response_t game_data{};
  auto response = Request<rc_api_fetch_game_data_request_t, rc_api_fetch_game_data_response_t>(
        fetch_data_request, &game_data, rc_api_init_fetch_game_data_request,
        rc_api_process_fetch_game_data_response);
  if (response == ResponseType::SUCCESS) {
    std::lock_guard lg{m_lock};
    m_game_data = game_data;
  }
  return response;
```


Also technically preexisting so you don't have to solve this in this PR, but aren't you leaking memory here? I would expect a call to `rc_api_destroy_fetch_game_data_response` *somewhere* but I don't see one.

Reply:
> Also technically preexisting so you don't have to solve this in this PR, but aren't you leaking memory here? I would expect a call to `rc_api_destroy_fetch_game_data_response` _somewhere_ but I don't see one.

This I'd need to ask around because I'm _pretty sure_ I tried doing this in the past and rcheevos doesn't seem to like initializing a response that's already been destroyed, which is a problem specifically for the game data response because it's the only one that's directly a member field in AchievementManager. I'll look into it and update accordingly.

Reply:
> This is a network request under the lock and will block everyone else until the request is done.

Now that we've talked about this being a problem with the fetch images, I do need to go back and correct all of the other network calls for this, but I would suggest putting it in a different PR entirely to keep this one clean. I'll work on that and get that up as soon as it's done.

Reply:
See https://github.com/dolphin-emu/dolphin/pull/12204